packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/ConfigurationImpl.kt

```diff
@@ -107,10 +107,12 @@ public open class ConfigurationImpl constructor(
 
     override suspend fun openRealm(realm: RealmImpl): Pair<FrozenRealmReference, Boolean> {
         val configPtr = realm.configuration.createNativeConfiguration()
-        val (dbPointer, fileCreated) = RealmInterop.realm_open(configPtr)
+        val scheduler = RealmInterop.realm_create_scheduler()
+        val (dbPointer, fileCreated) = RealmInterop.realm_open(configPtr, scheduler)
         val liveRealmReference = LiveRealmReference(realm, dbPointer)
         val frozenReference = liveRealmReference.snapshot(realm)
         liveRealmReference.close()
+        scheduler.release()
```

Comment:
We will leak this if `realm_open` throws. Also, how do we feel about having to close this manually and not as part of `Realm.close()`, but unfortunately when setting up the LiveRealmReference this way, it doesn't seem like that is possible.